### PR TITLE
feat(settings): add rescripts, allow external imports

### PR DESCRIPTION
**Pull request by jodyheavener**
Sort of an extension to #4819, but also a precursor for #4817 

This PR allows `fxa-settings` to import React components beyond its `./src` directory. This is not allowed in a standard CRA setup, so we are introducing [Rescripts](https://github.com/harrysolovay/rescripts) per [ADR 0013](https://github.com/mozilla/fxa/blob/master/docs/adr/0013-react-toolchain-for-settings-redesign.md#decision-outcome) to allow us to override CRA's Webpack configuration.

I've documented what I'm doing in `.rescriptsrc.js` -- it might _look_ scary, but it's actually pretty straightforward.

### CODE REVIEW

**Comment by jodyheavener** [comment]
cool tests bro

**Comment by dannycoates** [comment]
I don't think fxa-shared is the right place for components. IMO it should be in it's own thing `fxa-components` maybe? I can be convinced otherwise, but shared already needs some housekeeping. This might be worth a separate thread, but I think it makes sense to at least have shared code meant exclusively for the backend in it's own package.

File: packages/fxa-settings/.rescriptsrc.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  //           },
+  //         ]
+  //       }
+  //     ]
+  //   }
+  // }
+  if (
+    config.module.rules[2] &&
+    config.module.rules[2].hasOwnProperty("oneOf") &&
+    config.module.rules[2].oneOf[1].test.toString() ===
```
Comment (dannycoates) [inline packages/fxa-settings/.rescriptsrc.js:42]: This is fine. If path churn here becomes a problem maybe [jsonpath](https://www.npmjs.com/package/jsonpath) could help.

**Comment by jodyheavener** [comment]
@LZoog just want to make sure you get eyes on this, since we talked about fxa-shared previously (it's going in fxa-components now). I also added an additional layer of checking in Rescripts since you last looked at that code in my demo.

**Pull request merged**